flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/component/FileJobGraphRetriever.java

```diff
@@ -45,26 +51,39 @@
 	@Nonnull
 	private final String jobGraphFile;
 
-	public FileJobGraphRetriever(@Nonnull String jobGraphFile) {
+	public FileJobGraphRetriever(@Nonnull String jobGraphFile, @Nullable File usrLibDir) throws IOException {
+		super(usrLibDir);
 		this.jobGraphFile = jobGraphFile;
 	}
 
 	@Override
 	public JobGraph retrieveJobGraph(Configuration configuration) throws FlinkException {
-		File fp = new File(jobGraphFile);
+		final File fp = new File(jobGraphFile);
 
 		try (FileInputStream input = new FileInputStream(fp);
 			ObjectInputStream obInput = new ObjectInputStream(input)) {
-
-			return (JobGraph) obInput.readObject();
+			final JobGraph jobGraph = (JobGraph) obInput.readObject();
+			addUserClassPathsToJobGraph(jobGraph);
+			return jobGraph;
 		} catch (FileNotFoundException e) {
 			throw new FlinkException("Could not find the JobGraph file.", e);
 		} catch (ClassNotFoundException | IOException e) {
 			throw new FlinkException("Could not load the JobGraph from file.", e);
 		}
 	}
 
-	public static FileJobGraphRetriever createFrom(Configuration configuration) {
-		return new FileJobGraphRetriever(configuration.getString(JOB_GRAPH_FILE_PATH));
+	private void addUserClassPathsToJobGraph(JobGraph jobGraph) {
+		final List<URL> classPaths = new ArrayList<>();
+
+		if (jobGraph.getClasspaths() != null) {
```

Comment:
FYI we can address this null checker in FLINK-14839. It is beyond the scope of this pr, but you can take a look at that JIRA since I think you will get the idea there.

Reply:
Yes. It cound. After that pr we could remove the checker. Thank you for letting me know.